libraries/standard/mqtt/include/mqtt.h

```diff
@@ -145,13 +145,30 @@ MQTTStatus_t MQTT_Init( MQTTContext_t * pContext,
                         const MQTTFixedBuffer_t * pNetworkBuffer );
 
 /**
- * @brief Establish a MQTT session.
+ * @brief Establish an MQTT session.
+ *
+ * This function will send MQTT CONNECT packet and receive a CONNACK packet.
+ * The send and receive from the network is through the transport interface.
+ * The maximum wait for receiving a CONNACK is managed in 2 ways.
+ * 1. If #timeoutMs is greater than 0:
+ *    Parameter #timeoutMs will be used for a #getTime based maximum timeout
+ *    for the network read for CONNACK packet.
+ * 2. If #timeoutMs is 0:
+ *    A network read would be retried upto the number of times configured by
+ *    #MQTT_MAX_CONNACK_RECEIVE_RETRY_COUNT for the CONNACK packet.
  *
  * @brief param[in] pContext Initialized MQTT context.
  * @brief param[in] pConnectInfo MQTT CONNECT packet parameters.
  * @brief param[in] pWillInfo Last Will and Testament. Pass NULL if not used.
- * @brief param[in] timeoutMs Timeout in milliseconds for receiving
- * CONNACK packet.
+ * @brief param[in] timeoutMs Timeout in milliseconds for receiving CONNACK
```

Comment:
This explanation is not clear and hand-wavy. In what conditions this can spend more time? What is correct configuration? What is the low probability condition?

Reply:
Also, I think it's a little odd this is tagged with a `@brief`, when the description is anything but

Reply:
Ok Thanks. Will reword it.